### PR TITLE
Feature/lab 4 numeronym

### LINKED ISSUE #460: Lab 04 - Numeroyms
# Lab 4 - Numeronym

- Create an executable go program in directory `04_numeronym/USERNAME`
- Write a function that returns a slice of numeronyms for its input strings:

```
func numeronyms(vals ...string) []string
```

Call `fmt.Println(numeronyms("accessibility", "Kubernetes", "abc"))` in `main` to print:

```
[a11y K8s abc]
```

### PR DESCRIPTION
**Pull request by starfallz**
Fixes #460

Review of colleague's PR #

#### Changes proposed in this PR:

- Write a function that returns a slice of numeronyms for its input strings

### CODE REVIEW

**Review by runnerdave — APPROVED** [review]
👍

File: 04_numeronym/starfallz/main_test.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+		description string
+		input       []string
+		expected    []string
+	}{
+		{"Test numeronyms with one string input", []string{"accessibility"}, []string{"a11y"}},
+		{"Test numeronyms with multiple string input", []string{"accessibility", "Kubernetes", "abc"},
+			[]string{"a11y", "K8s", "abc"}},
+		{"Test numeronyms with nil to return empty slice", nil, []string{}},
+		{"Test numeronyms with empty to return empty slice", []string{}, []string{}},
+		{"Test numeronyms with empty string to return empty string", []string{""}, []string{""}},
```
Comment (juliaogris) [inline 04_numeronym/starfallz/main_test.go:38]: Please add tests with non-ascii unicode characters, e.g. emojis etc.

**Review by juliaogris — CHANGES_REQUESTED** [review]
Please add link to review of colleague's PR.
Please [fix commit messages](https://github.com/anz-bank/go-course/wiki/Lab-1:-Collected-Code-Review-Comments#commit-messages) with `git rebase -i HASH`

File: 04_numeronym/starfallz/main.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+)
+
+const minNumberOfLetters = 3
```
Comment (alextmz) [inline 04_numeronym/starfallz/main.go:11]: Although I do like the visibility of the variable being pushed here, I'm also divided between visibility and avoiding the use of global state. I'd say go with the latter unless you know of a better guidance.

File: 04_numeronym/starfallz/main_test.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+	for _, testCase := range testCases {
+		input := testCase.input
+		expected := testCase.expected
+
+		t.Run(testCase.description, func(t *testing.T) {
+			actual := numeronyms(input...)
+
+			if !reflect.DeepEqual(actual, expected) {
+				t.Errorf("Unexpected output, expected: %s, actual: %s", expected, actual)
```
Comment (alextmz) [inline 04_numeronym/starfallz/main_test.go:49]: Do not uppercase  error strings (https://github.com/golang/go/wiki/CodeReviewComments#error-strings)

File: 04_numeronym/starfallz/main_test.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+		t.Run(testCase.description, func(t *testing.T) {
+			actual := convertString(input)
+
+			if expected != actual {
+				t.Errorf("Unexpected output, expected: %s, actual: %s", expected, actual)
+			}
+		})
+	}
+}
```
Comment (alextmz) [inline 04_numeronym/starfallz/main_test.go:82]: Overall nice code! Clean and concise.

**Review by alextmz — COMMENTED** [review]
Some minor changes suggested
